### PR TITLE
feat(gyp): update gyp to v0.15.0

### PR DESCRIPTION
**Pull request by rzhao271**
##### Checklist

- [x] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/googleapis/release-please#how-should-i-write-my-commits)

##### Description of change

Update gyp to v0.15.0. Brings in SpectreMitigation flag support for MSVS along with some bug fixes and doc improvements.

### CODE REVIEW

File: gyp/.github/workflows/Python_tests.yml [hunk 1]
```diff
@@ -30,7 +30,9 @@ jobs:
       - run: ./gyp -V && ./gyp --version && gyp -V && gyp --version
       - name: Lint with flake8
         run: flake8 . --ignore=E203,W503  --max-complexity=101 --max-line-length=88 --show-source --statistics
-      - name: Test with pytest
+      - name: Lint with ruff  # See pyproject.toml for settings
+        run: flake8 .
```
Comment (cclauss) [inline gyp/.github/workflows/Python_tests.yml:34]:
```suggestion
      - name: Lint with ruff  # See pyproject.toml for settings
        run: ruff .
```

**Comment by rzhao271** [comment]
Looks like there's more linting errors.
I'm wondering whether the errors should be fixed on the gyp-next side first?

**Comment by cclauss** [comment]
;-) https://github.com/nodejs/gyp-next/pull/193

File: gyp/pylib/gyp/xcodeproj_file.py [hunk 2]
```diff
@@ -2770,7 +2769,7 @@ def __init__(self, properties=None, id=None, parent=None, path=None):
         self.path = path
         self._other_pbxprojects = {}
         # super
-        XCContainerPortal.__init__(self, properties, id, parent)
+        return XCContainerPortal.__init__(self, properties, id, parent)
```
Comment (cclauss) [inline gyp/pylib/gyp/xcodeproj_file.py:2772]: Please revert this change to placate the linter (which is correct).

**Comment by rzhao271** [comment]
CI tests fail, but I wonder whether it's from 02480f6b6894a6bb81dd3e1365bc4df6d5336f7c? That commit's CI status looks similar to this PR's.

**Comment by cclauss** [comment]
Please rebase this pull request.

**Comment by joaomoreno** [comment]
@cclauss What can we do to get this rolling?

File: gyp/README.md [hunk 3]
```diff
@@ -5,3 +5,26 @@ Documents are available at [gyp.gsrc.io](https://gyp.gsrc.io), or you can check
 
 __gyp-next__ is [released](https://github.com/nodejs/gyp-next/releases) to the [__Python Packaging Index__](https://pypi.org/project/gyp-next) (PyPI) and can be installed with the command:
 * `python3 -m pip install gyp-next`
+
+When used as a command line utility, __gyp-next__ can also be installed with [pipx](https://pypa.github.io/pipx):
+* `pipx install gyp-next`
+```
+Installing to a new venv 'gyp-next'
+  installed package gyp-next 0.13.0, installed using Python 3.10.6
```
Comment (cclauss) [inline gyp/README.md:13]:
```suggestion
  installed package gyp-next 0.13.0, installed using Python 3.12.0
```

File: gyp/pyproject.toml [hunk 4]
```diff
@@ -20,22 +20,39 @@ classifiers = [
...
     "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
 ]
 
 [project.optional-dependencies]
-dev = ["flake8", "pytest"]
+dev = ["flake8", "ruff", "pytest"]
```
Comment (cclauss) [inline gyp/pyproject.toml:31]:
```suggestion
dev = ["ruff", "pytest"]
```

File: gyp/pyproject.toml [hunk 5]
```diff
@@ -20,22 +20,39 @@ classifiers = [
...
+    "C9",
+    "E",
+    "F",
+    "PLC",
+    "PLE",
+    "PLW",
+    "Q",
+    "W",
+ ]
+ target-version = "py37"
```
Comment (cclauss) [inline gyp/pyproject.toml:51]:
```suggestion
 target-version = "py38"
```

**Comment by cclauss** [comment]
I think we will want to get Python 3.12 support in this release.
* nodejs/gyp-next#214

**Comment by rzhao271** [comment]
Closing in favour of https://github.com/nodejs/node-gyp/pull/2923
